userspace/libsinsp/sinsp.cpp

```diff
@@ -974,8 +974,26 @@ void sinsp::on_new_entry_from_proc(void* context,
 		{
 			if(m_filter != nullptr && is_capture())
 			{
-				// we'll run the filter when we see the fds and possibly clear the filtered_out flag
-				newti->m_filtered_out = true;
+				scap_evt tscapevt;
+				tscapevt.ts = 0;
+				tscapevt.type = PPME_SYSCALL_READ_X;
+				tscapevt.len = 0;
+				tscapevt.nparams = 0;
+				tscapevt.tid = newti->m_tid;
+
+				sinsp_evt tevt;
+				tevt.m_pevt = &tscapevt;
+				tevt.m_inspector = this;
+				tevt.m_info = &(g_infotables.m_event_info[PPME_SYSCALL_READ_X]);
+				tevt.m_cpuid = 0;
+				tevt.m_evtnum = 0;
+				tevt.m_tinfo = newti;
+				tevt.m_fdinfo = NULL;
+				tevt.m_tinfo->m_lastevent_fd = -1;
+
+				newti->m_filtered_out = m_filter->run(&tevt);
+				newti->m_lastevent_fd = -1;
+				newti->m_lastevent_data = NULL;
```

Comment:
What fd is this going to use for the fake read event?

Reply:
I thought that setting `newti->m_lastevent_fd` as `-1` was enough to disable all possible fd-related filterchecks and have `<NA>` values. My thought process was that `PPME_SYSCALL_READ_X` does not carry any fd info (thus `extract_fd` will fail and fd filterchecks will break-out with NULL), and that resetting `m_lastevent_fd` to `-1` would make all other code paths consider the fs as missing. Totally open to better ideas if you have more insights!

Reply:
Depending on what we're trying to achieve here and with how much effort, I feel like the least surprising way would be to:
* mark everything as filtered out
* clear the flag on the thread/fd whenever an actual scap event matches the filter

Unfortunately, this would require two passes over the scap file (the proc/fd lists are written at the beginning of the capture, before any events). Still, it seems like the only way to have a filter like 'evt.type=execve' (or anything except read) work.

Thoughts?